bbpower/compsep.py

```diff
@@ -611,7 +611,7 @@ def emcee_sampler(self):
         import emcee
         from multiprocessing import Pool
 
-        fname_temp = self.get_output('param_chains')+'.h5'
+        fname_temp = '/'.join(self.get_output('param_chains').split('/')[:-1])+'/emcee.npz.h5'
```

Comment:
Can you explain what's going on here?

Reply:
We have several samplers in the pipeline (emcee, polychord, predicted_spectra etc.) that we might want to use on the same simulation but in the original code all outputs write to the same file (called by self.get_output('param_chains')). This patch solved the issue by just taking the leaf directory and adding a new name for each sampler, but it could be done in a much cleaner way, eg by reading just the output path from config.